static/src/javascripts/projects/common/modules/crosswords/helpers.js

```diff
@@ -0,0 +1,360 @@
+// @flow
+import { constants } from 'common/modules/crosswords/constants';
+import findIndex from 'lodash/arrays/findIndex';
+import flatten from 'lodash/arrays/flatten';
+import range from 'lodash/arrays/range';
+import uniq from 'lodash/arrays/uniq';
+
+const isAcross = (clue: Clue): boolean => clue.direction === 'across';
+
+const getLastCellInClue = (clue: Clue): Position => {
+    const ax = {
+        true: 'x',
+        false: 'y',
+    };
+    const axis = ax[String(isAcross(clue))];
+    const otherAxis = ax[String(!isAcross(clue))];
+    const cell: Position = ({
+        [axis]: clue.position[axis] + (clue.length - 1),
+        [otherAxis]: clue.position[otherAxis],
+    }: any);
+
+    return cell;
+};
+
+const isFirstCellInClue = (cell: Position, clue: Clue): boolean => {
+    const axis = isAcross(clue) ? 'x' : 'y';
+
+    return cell[axis] === clue.position[axis];
+};
+
+const isLastCellInClue = (cell: Position, clue: Clue): boolean => {
+    const axis = isAcross(clue) ? 'x' : 'y';
+
+    return cell[axis] === clue.position[axis] + (clue.length - 1);
+};
+
+const getNextClueInGroup = (entries: Array<Clue>, clue: Clue): ?Clue => {
+    const newClueId =
+        clue.group[findIndex(clue.group, id => id === clue.id) + 1];
```

Comment:
`Array.prototype.findIndex` behaves in the same way as lodash `findIndex` (returns the index of the first element that passes the callback check), so this could instead be written as `clue.group.findIndex(id => id === clue.id)` and remove this lodash lib?